src/app/test_executive/test_executive.ml

```diff
@@ -67,54 +67,68 @@ let main inputs =
   let net_manager_ref : Engine.Network_manager.t option ref = ref None in
   let log_engine_ref : Engine.Log_engine.t option ref = ref None in
   let cleanup_deferred_ref = ref None in
-  let dispatch_cleanup reason test_error =
-    let open Deferred.Let_syntax in
-    let cleanup test_error =
-      let test_error_set =
-        match test_error with
-        | None ->
-            Test_error.Set.empty
-        | Some err ->
-            Test_error.Set.hard_singleton (Test_error.internal_error err)
-      in
-      let%bind log_engine_error_set =
-        Option.value_map !log_engine_ref
-          ~default:(Deferred.return Test_error.Set.empty) ~f:(fun log_engine ->
-            match%map Engine.Log_engine.destroy log_engine with
-            | Ok errors ->
-                errors
-            | Error err ->
-                Test_error.Set.hard_singleton (Test_error.internal_error err)
-        )
-      in
-      let%bind () =
-        Option.value_map !net_manager_ref ~default:Deferred.unit
-          ~f:Engine.Network_manager.cleanup
+  let dispatch_cleanup reason (test_error : 'a Malleable_error.t) =
+    let cleanup () =
+      let test_error_with_cleanup_result =
+        let open Malleable_error.Let_syntax in
+        let%bind res =
+          let open Deferred.Let_syntax in
+          match%bind test_error with
```

Comment:
Just use `Malleable_error.lift_error_set` instead. This is basically just a reimplementation of that function. Also, you can do this outside of this expression and combine it with the logic in line 108, but that's optional.

Reply:
but the code here wraps it differently than lift_error_set doesn't it?  

the code here uses Malleable_error.return no matter if test_error is Ok or Error.  the Set gets thereby wrapped as the "computation result" of the Malleable_error.  

on the other hand using lift_error_set doesn't actually return a Malleable error, it uses the ocaml Result thing but it's not a malleable error and i'm not sure what to do with it.